content/en/tracing/connect_logs_and_traces/_index.md

```diff
@@ -439,23 +474,66 @@ Coming Soon. Reach out to [the Datadog support team][1] to learn more.
 {{% /tab %}}
 {{% tab ".NET" %}}
 
-To manually inject trace identifiers into your logs, access the necessary values through the `CorrelationIdentifier` static class. If your logging library supports structured logging, such as JSON messages, add the `dd.trace_id` and `dd.span_id` properties with their respective values.
+If you prefer to manually correlate your [traces][1] with your logs, leverage the Datadog API to retrieve correlation identifiers:
 
-Otherwise, add the strings `dd.trace_id=<TRACE_ID>` and `dd.span_id=<SPAN_ID>` to your log message. For example:
+* Use `CorrelationIdentifier.TraceId` and `CorrelationIdentifier.SpanId` API methods to inject identifiers at the beginning and end of each [span][2] to log (see examples below).
+* Configure MDC to use the injected Keys:
+  * `dd.trace_id` Active Trace ID during the log statement (or `0` if no trace)
+  * `dd.span_id` Active Span ID during the log statement (or `0` if no trace)
+* `Serilog` example:
 
 ```csharp
 using Datadog.Trace;
+using Serilog.Context;
+
+// there must be spans started and active before this block.
+using (LogContext.PushProperty("dd.trace_id", CorrelationIdentifier.TraceId.ToString()))
+using (LogContext.PushProperty("dd.span_id", CorrelationIdentifier.SpanId.ToString()))
+{
+    // Log something
+}
+```
+
+* `NLog` example:
 
-var traceId = CorrelationIdentifier.TraceId;
-var spanId = CorrelationIdentifier.SpanId;
+```csharp
+using Datadog.Trace;
+using NLog;
 
-var message = $"My log message. [dd.trace_id={traceId} dd.span_id={spanId}]";
+// there must be spans started and active before this block.
+using (MappedDiagnosticsLogicalContext.SetScoped("dd.trace_id", CorrelationIdentifier.TraceId.ToString()))
+using (MappedDiagnosticsLogicalContext.SetScoped("dd.span_id", CorrelationIdentifier.SpanId.ToString()))
+{
+    // Log something
+}
 ```
 
-**Note**: If you are not using a [Datadog Log Integration][1] to parse your logs, custom log parsing rules need to ensure that `trace_id` and `span_id` are being parsed as a string. More information can be found in the [FAQ on this topic][2].
+* `log4net` example:
 
-[1]: /logs/log_collection/csharp/#configure-your-logger
-[2]: /tracing/faq/why-cant-i-see-my-correlated-logs-in-the-trace-id-panel
+```csharp
+using Datadog.Trace;
+using log4net;
+
+// there must be spans started and active before this block.
+try
+{
+    LogicalThreadContext.Properties["dd.trace_id"] = CorrelationIdentifier.TraceId.ToString();
+    LogicalThreadContext.Properties["dd.span_id"] = CorrelationIdentifier.SpanId.ToString();
```

Comment:
I wonder if there is a way to get the lines within the `try` block to be indented.

Reply:
I think there is an issue with our Java and C# markdown renderers because this happens with Java too 😢